GameClient/Assets/Code/Game/ClientPlayer/PlayerBehaviour.cs

```diff
@@ -20,5 +22,18 @@ public override void OnBeforeMoveTile(Position movingTo)
             UID = UnityClient.Player.UID,
             To = movingTo
         });
+        
+        var chunkX = UnityClient.Player.Position.X >> 4;
+        var chunkY = UnityClient.Player.Position.Y >> 4;
+
+        var chunkActivedParents = PositionExtensions.GetSquared3x3Around(new Position(chunkX, chunkY)).ToList();
+        var chunkActivedParentsNames = new List<string>();
+        chunkActivedParents.ForEach(c => chunkActivedParentsNames.Add($"chunk_{c.X}_{c.Y}"));
+
+        foreach (var cp in UnityClient.Map.Chunks.Values)
```

Comment:
Isso num mapa infinito pode dar um pipoco se pegar um player muito andarilho huasehauhse pq vai percorrer todos chunks q o client ja carregou, fora q isso ta rodando sempre q o player move, mesmo se for no mesmo chunk, rola botar uma verificacao se ele mudou de chunk.

Pra nao loopar todos os chunks, vc pode pegar o 3x3 q ele tava e o 3x3 q ele tava indo. Oq tiver na primeira lista e nao na segunda, inativa.

Reply:
Verdade, não tinha pensado nisso, vou realizar as alterações aqui o/